tests/integration-tests/conftest.py

```diff
@@ -167,7 +167,12 @@ def clusters_factory(request):
 
     def _cluster_factory(cluster_config):
         if request.config.getoption("cluster"):
-            cluster = Cluster(name=request.config.getoption("cluster"), ssh_key=request.config.getoption("key_path"))
+            cluster_config = _write_cluster_config_to_outdir(request, cluster_config)
```

Comment:
Minor: if and else branches differ only for the cluster name

Reply:
Changed 👍